learning_journal/views/default.py

```diff
@@ -1,20 +1,93 @@
 from pyramid.response import Response
 from pyramid.view import view_config
+from pyramid.httpexceptions import HTTPFound
+from pyramid.httpexceptions import exception_response
+from pyramid.security import remember, forget
+from learning_journal.security import check_credentials
+
+
+import time
+import datetime
 
 from sqlalchemy.exc import DBAPIError
 
-from ..models import MyModel
+from ..models import Entry
+
+
+@view_config(route_name='login', renderer='../templates/login.jinja2', permission="view")
+def login_view(request):
+    """Handle the login view."""
+    if request.POST:
+        username = request.POST["username"]
+        password = request.POST["password"]
+        if check_credentials(username, password):
+            auth_head = remember(request, username)
+            return HTTPFound(
+                location=request.route_url("homepage"),
+                headers=auth_head
+            )
+
+    return {}
 
 
-@view_config(route_name='home', renderer='../templates/mytemplate.jinja2')
+@view_config(route_name='logout')
+def logout_view(request):
+    """Handle logging the user out."""
+    headers = forget(request)
+    return HTTPFound(request.route_url("homepage"), headers=headers)
+
+
+@view_config(route_name='homepage', renderer='../templates/index.jinja2', permission="view")
 def my_view(request):
+    """View for homepage, listing journal entries from database."""
     try:
-        query = request.dbsession.query(MyModel)
-        one = query.filter(MyModel.name == 'one').first()
+        entries = request.dbsession.query(Entry).all()
     except DBAPIError:
         return Response(db_err_msg, content_type='text/plain', status=500)
-    return {'one': one, 'project': 'learning_journal'}
+    return {'entries': entries}
+
+
+@view_config(route_name="write", renderer="../templates/write.jinja2", permission="amend")
+def write(request):
+    if request.method == "POST":
+        if request.POST["title"] and request.POST["body"]:
+            new_title = request.POST["title"]
+            new_body = request.POST["body"]
+            new_date = datetime.date.today()
+            new_entry = Entry(title=new_title, body=new_body, creation_date=new_date)
 
+            request.dbsession.add(new_entry)
+            return HTTPFound(location=request.route_url('homepage'))
+        else:
+            return HTTPFound(location=request.route_url('write'))
```

Comment:
This `else` block is unnecessary. No matter what you're going back to the same page.